source/benchmarks/memory_benchmark/implementations/ocl/write_buffer_ocl.cpp

```diff
@@ -44,13 +44,13 @@ static TestResult run(const WriteBufferArguments &arguments, Statistics &statist
     HostptrReuseHelper::Alloc hostptrAlloc{};
     ASSERT_CL_SUCCESS(HostptrReuseHelper::allocateBufferHostptr(opencl, arguments.reuse, arguments.size, hostptrAlloc));
 
+    ASSERT_CL_SUCCESS(BufferContentsHelperOcl::fillBuffer(opencl.commandQueue, buffer, arguments.size, arguments.contents));
+
     // Warmup
     ASSERT_CL_SUCCESS(clEnqueueWriteBuffer(opencl.commandQueue, buffer, CL_BLOCKING, 0, arguments.size, hostptrAlloc.ptr, 0, nullptr, nullptr));
 
     // Benchmark
     for (auto i = 0u; i < arguments.iterations; i++) {
-        ASSERT_CL_SUCCESS(BufferContentsHelperOcl::fillBuffer(opencl.commandQueue, buffer, arguments.size, arguments.contents));
```

Comment:
Won't this defeat the purpose of `arguments.contents`? The buffer will have specified contents only for warmup run, but not for actually measured runs.

Reply:
Yeah contents doesn't make a lot sense in this benchmark , as prime thing that we want to meassure here is how write buffer performs in "cold" enironment. By introducing fill in every iteration it does side effects which are not desired.